library/std/src/panicking.rs

```diff
@@ -196,6 +196,8 @@ fn default_hook(info: &PanicInfo<'_>) {
         Some(s) => *s,
         None => match info.payload().downcast_ref::<String>() {
             Some(s) => &s[..],
+            #[cfg(not(bootstrap))]
+            None if info.error().is_some() => "non-recoverable runtime Error",
```

Comment:
Once `std::error::Report` is merged this will be updated to actually report the source error when one exists, rather than requiring the user installs a custom hook that actually prints the error.